### ISSUE TITLE
error.sourceName must not be null

### ISSUE THREAD

**Issue by douglasbandeiraivo** [gradle-webtools/gradle-minify-plugin#25]
I'm trying to use the plugin, but I'm getting an error. 

My code in _build.gradle_ is:
```
plugins {
    id 'com.android.application'
    id "org.gradlewebtools.minify" version "1.2.1"
}
...
task jsMinify(type: org.gradlewebtools.minify.JsMinifyTask) {
    srcDir = project.file("$projectDir/src/main/assets/Token/app")
    dstDir = project.file("$projectDir/src/main/assets/Token/appmin")
    options.originalFileNames = true
    options.compilationLevel = com.google.javascript.jscomp.CompilationLevel.SIMPLE_OPTIMIZATIONS
}
```

The error I'm getting is:

```
Execution failed for task ':app:jsMinify'.
> error.sourceName must not be null

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:jsMinify'.
Caused by: java.lang.NullPointerException: error.sourceName must not be null
	at org.gradlewebtools.minify.minifier.result.Event.<init>(Event.kt:24)
	at org.gradlewebtools.minify.minifier.result.Error.<init>(Error.kt:8)
	at org.gradlewebtools.minify.minifier.js.JsMinifier.minifyFile(JsMinifier.kt:40)
	at org.gradlewebtools.minify.minifier.Minifier.minifyFileInternal(Minifier.kt:76)
	at org.gradlewebtools.minify.minifier.Minifier.handleFile(Minifier.kt:69)
	at org.gradlewebtools.minify.minifier.Minifier.minifyInternal(Minifier.kt:40)
	at org.gradlewebtools.minify.minifier.Minifier.minify(Minifier.kt:28)
	at org.gradlewebtools.minify.JsMinifyTask.minify(JsMinifyTask.kt:25)
```

Anyone could help me to bypass the problem?

**Comment by timon-schelling** [comment]
Hi, thanks for using the plugin. 

It seems like something in you're JS code somehow leads to the Closure Compiler (the tool this plugin uses for JS minification and optimization https://developers.google.com/closure/compiler) returning an error with no "sourceName" specified, but our code expects "sourceName" to never be null. 

In the commit above, I added a workaround that should help for now. But another fix is needed in the long run. @616slayer616 should we open a new issue for that?  

I hope a release with the workaround will be available soon.

Could you provide us with the JS code that leads to that unexpected behavior?

**Comment by 616slayer616** [comment]
@timon-schelling 
Very nice that you could fix it so fast.
Do you have an idea for a more extensive solution of the problem. And a name / description for it to create a new issue?
If so I a new issue would be nice.

**Comment by timon-schelling** [comment]
@douglasbandeiraivo release 1.2.2 (https://plugins.gradle.org/plugin/org.gradlewebtools.minify/1.2.2) should fix your problem, but keep in mind that there might be another error in your JS, otherwise there would be no JSError.

**Comment by timon-schelling** [comment]
I will keep this open until there is a new issue for a permanent fix.

**Comment by douglasbandeiraivo** [comment]
Hi, @timon-schelling. Thanks for the quick reply. I discovered the problem was with the following line of code, probably an encoding issue. When I removed the special character, this problem was resolved.

`alert("QR Code Inválido");`

**Comment by timon-schelling** [comment]
> Hi, @timon-schelling. I discovered the problem was with the following line of code, probably an encoding issue. When I removed the special character, this problem was resolved.
> 
> `alert("QR Code Inválido");`
> 

And what error is shown?

**Comment by 616slayer616** [comment]
Weird I cannot recreate the issue with the special character:
https://github.com/gradle-webtools/gradle-minify-plugin/tree/issue_25

No warning or error is shown even after I reverted the fix.

@douglasbandeiraivo can you check if you get the issue if this is the only line of code in the file?

**Comment by douglasbandeiraivo** [comment]
> > Hi, @timon-schelling. I discovered the problem was with the following line of code, probably an encoding issue. When I removed the special character, this problem was resolved.
> > `alert("QR Code Inválido");`
> 
> And what error is shown?

error.sourceName must not be null

**Comment by douglasbandeiraivo** [comment]
> Weird I cannot recreate the issue with the special character:
> https://github.com/gradle-webtools/gradle-minify-plugin/tree/issue_25
> 
> No warning or error is shown even after I reverted the fix.
> 
> @douglasbandeiraivo can you check if you get the issue if this is the only line of code in the file?

I'm not with access to the PC now. I'll try this later.

**Comment by timon-schelling** [comment]
> > > Hi, @timon-schelling. I discovered the problem was with the following line of code, probably an encoding issue. When I removed the special character, this problem was resolved.
> > > `alert("QR Code Inválido");`
> > 
> > And what error is shown?
> 
> error.sourceName must not be null

Sorry, I meant with the fix.

**Comment by douglasbandeiraivo** [comment]
> > > > Hi, @timon-schelling. I discovered the problem was with the following line of code, probably an encoding issue. When I removed the special character, this problem was resolved.
> > > > `alert("QR Code Inválido");`
> > > 
> > > 
> > > And what error is shown?
> > 
> > 
> > error.sourceName must not be null
> 
> Sorry, I meant with the fix.

With the fix, the error message changed and now points a possible problem with the encoding (as I had supposed before).

The error message is:

> > Task :app:jsMinify
> ERROR - [JSC_READ_ERROR] Cannot read file C:/.../app/app.js: Failed to read: C:\...\app\app.js, is this input UTF-8 encoded?
> 
> 1 error(s), 0 warning(s)
> Error: Cannot read file C:/.../app/app.js: Failed to read: C:\...\app\app.js, is this input UTF-8 encoded? at  line -1:-1
> Cannot read file C:/.../app/app.js: Failed to read: C:\...\app\app.js, is this input UTF-8 encoded? at  line -1:-1
> 
> > Task :app:jsMinify FAILED
>

**Comment by douglasbandeiraivo** [comment]
> @douglasbandeiraivo can you check if you get the issue if this is the only line of code in the file?

@616slayer616, the problem also occurs when I put just this line of code. I attached the file to you simulate the problem.

[app.zip](https://github.com/gradle-webtools/gradle-minify-plugin/files/6897370/app.zip)

**Comment by douglasbandeiraivo** [comment]
@timon-schelling, saving the file with UTF-8 encoding also resolved the problem.

**Comment by 616slayer616** [comment]
@douglasbandeiraivo Thank you for the support.
I could recreate the issue here: https://github.com/gradle-webtools/gradle-minify-plugin/tree/issue_25.

But I don't know if we can do anything about that since the error occurs in the closure compiler.

I would recommend saving files in UTF-8

**Comment by douglasbandeiraivo** [comment]
> @douglasbandeiraivo Thank you for the support.
> I could recreate the issue here: https://github.com/gradle-webtools/gradle-minify-plugin/tree/issue_25.
> 
> But I don't know if we can do anything about that since the error occurs in the closure compiler.
> 
> I would recommend saving files in UTF-8

Ok, @616slayer616. Thanks.

**Comment by 616slayer616** [comment]
@timon-schelling 
I think we can clots this?

**Comment by timon-schelling** [comment]
Yes, maybe with a link to the fix.

**Comment by 616slayer616** [comment]
Fixed in [1.2.2](https://github.com/gradle-webtools/gradle-minify-plugin/releases/tag/v1.2.2)

**Comment by 616slayer616** [comment]
Fixed in [1.2.2](https://github.com/gradle-webtools/gradle-minify-plugin/releases/tag/v1.2.2)
